Identity/Identity.API/Controllers/IdentityController.cs

```diff
@@ -0,0 +1,154 @@
+using Identity.DataAccess;
+using Identity.DataAccess.Models;
+using Identity.DataAccess.Models.Entities;
+using Identity.Public;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Npgsql;
+using System.Security.Claims;
+
+namespace Identity.API.Controllers;
+
+[ApiController]
+public class IdentityController(IdentityDatabaseContext dbContext, JWTHelper jwtHelper) : ControllerBase
+{
+    [HttpGet]
+    [Route("users")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<IEnumerable<User>>> GetAllUsers()
+    {
+        IEnumerable<User> users;
+        try
+        {
+            users = (await dbContext.Users.ToListAsync())
+                .Select(u => new User() { Username = u.Username, Email = u.Email });
+        }
+        catch
+        {
+            return StatusCode(StatusCodes.Status500InternalServerError);
+        }
+
+        return Ok(users);
+    }
+
+    [HttpGet]
+    [Route("user/{id}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<User?>> GetUser(int id)
+    {
+        User user;  
+
+        try
+        {
+            UserEntity? userEntity = await dbContext.Users.FindAsync(id);
+
+            if (userEntity is null)
+                return NotFound();
+
+            user = new User
+            { 
+                Username = userEntity.Username,
+                Email = userEntity.Email
+            };
+        }
+        catch
+        {
+            return StatusCode(StatusCodes.Status404NotFound);
+        }
+
+        return Ok(user);
+    }
+
+    [Route("register")]
+    [HttpPost]
+    [ProducesResponseType(StatusCodes.Status201Created)]
+    [ProducesResponseType(StatusCodes.Status409Conflict)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<string>> RegisterUser(UserCreateRequest userCreateRequest)
+    {
+        UserEntity userToCreate = new UserEntity()
+        {
+            Username = userCreateRequest.Username,
+            PasswordHash = HashingHelper.HashPassword(userCreateRequest.Password),
+            Email = userCreateRequest.Email,
+            Roles = new List<string> { UserRoles.ADMIN, UserRoles.MEMBER },
+        };
+
+        try
+        {
+            await dbContext.Users.AddAsync(userToCreate);
+            await dbContext.SaveChangesAsync();
+        }
+        catch (DbUpdateException ex) when ((ex.InnerException as PostgresException)?.SqlState == "23505")
+        { 
+            return Conflict();
+        }
+        catch
+        {
+            return StatusCode(StatusCodes.Status500InternalServerError);
+        }
+
+        return Ok(jwtHelper.GenerateJwtToken(userToCreate));
```

Comment:
This currently returns 200 instead of 201. Check [this](https://ochzhen.com/blog/created-createdataction-createdatroute-methods-explained-aspnet-core#createdataction-string-actionname-object-routevalues-object-value) out